### ISSUE TITLE
[Feature] Allow 'spiral' to go clockwise or counter clockwise?

### ISSUE THREAD

**Issue by daantimmer** [Courseplay/Courseplay_FS22#2059]
**What is the problem this feature/enhancement solves?**
I might have missed it, or there is a current case of PEBCAK, but when I chose the 'spiral' layout it'll always go counterclockwise. Which means harvesters unload point is always on the crop's side.

I would like the spiral layout to go clockwise instead, which allows the unload trucks to not go through the unharvested crop field.

**Describe the solution you'd like**
Let the Spiral layout follow a (counter)clockwise setting, just like can be done with the headlands

**Describe alternatives you've considered**
None really. i tried the headlands direction setting, hoping it would work, but it did not :-(

**Additional context**
Not really.

**Comment by Tensuko** [comment]
> None really. i tried the headlands direction setting, hoping it would work, but it did not :-(

Headland should consider CW/CCW setting.

What do you try to do with the Spiral layout?

**Comment by daantimmer** [comment]
> Headland should consider CW/CCW setting.
The headland considers the CW/CCW settings. I was somewhat hoping the spiral setting would as well.

> What do you try to do with the Spiral layout?
I would like a layout where the Combine's Pipe is _always_ pointing outwards, not inwards.
But it seems the spiral layout is _always_ CCW, which means the pipe is always pointing to the inside of the field.

**Comment by pvaiko** [comment]
Have you tried the 'Lands' pattern? (with even number of rows in a land)

**Comment by Tensuko** [comment]
For Combines I can recommend you 2-3 Headlands (wich are outside the fruit if choose the right direction, pipe on the left ist CW).
And then as pvaiko said, Center pattern lands, try it with 6 for the start, for bigger fields a bigger numer works fine too.

**Comment by rohne83** [comment]
Spirals pattern is good for rather narrow fields where convoys of combines and/or active unloaders are working. Reason for the problem is, that pipe side isn't considered right now for the fieldrows of spiral courses. It currently only depends on where the start point and thus the transition from headland to center is, if the pipe on field rows is always outside the fruit or always inside. 

But this issue here is duplicate of #820

**Comment by daantimmer** [comment]
I shall have a look at the land pattern next harvest season! I shall update this issue if I get a better/different/other/good result

**Comment by cwattyeso** [comment]
I'm in the camp that there should be some way to specify to a harvester when doing spirals to ensure it does so with the pipe on the opposite side to the crop so that  unloaders have free access at all times to the pipe for unloading. As I tend to do harvesting in combination with Autodrive doing the unloading and transporting of the harvester, anything that can make life easier for the Autodrive workers would be appreciated, especially as that mod has quite a few issues still in the Harvesting/Loading Department and sadly doesn't look like it's ever going to get an update, with the last coming way back in July. I really need to have Courseplay provide as simple and easy method of harvesting fields so that other workers or autodrive can play much nicer with it.

**Comment by pvaiko** [comment]
@cwattyeso First of all, Courseplay has its own combine unloader back now.

And yes, I know it would be nice, and sounds like a trivial change, but the way the generator implemented now does not make this an easy task. I'll think about it. 

In the meanwhile, how about adding as much headlands as you can?

**Comment by cwattyeso** [comment]
Yeah I saw Courseplay has an Unloader option, but haven't yet played with this feature, because I've not seen any tutorials or instructions anywhere on how it works. I tried using it the other day on a map during a large field harvesting, but kept getting a message from the trailer when trying to start the combine unloader task that it couldn't because it was full or something.

**Comment by rohne83** [comment]
It's not a lottery, but strictly depending on the starting point of the course and the resulting headland transitions: look into the issue #820 where I depicted where you have to place the starting point to be sure already today, that the spirals field rows are worked in the correct direction.
Normally a setting shouldn't be needed at all. Just a pipe side logic (which already exists for the lands pattern) should be used as well for spirals pattern. But don't know how complicated it is to be integrated into the course generator for spirals.

**Comment by pvaiko** [comment]
> Yeah I saw Courseplay has an Unloader option, but haven't yet played with this feature, because I've not seen any tutorials or instructions anywhere on how it works.

See https://github.com/Courseplay/Courseplay_FS22/releases/tag/7.1.1.0

**Comment by pvaiko** [comment]
> to ensure the Harvester is doing the longest rows and not doing loads of short rows,
Courseplay has logic to find the direction which results in the lowest number of rows in auto mode, or for the longest, you could use 'longest edge'. But if all your fields are rectangular and oriented to the cardinal direction, it does not really matter...

**Comment by cwattyeso** [comment]
Will give it a try next time I'm in game. I'm guessing the reason it didn't work when I attempted it the other day was I was using normal Giants AI to drive my potato harvesters up and down the field, and you say the harvesters must be courseplay helpers. Also I was using a Truck and Trailer to unload, and you say it only works with Tractor and Trailer, so again when doing such contracts need to check the offered equipment first and adjust if needed to ensure I have tractors and trailers to do the unloading.



> > to ensure the Harvester is doing the longest rows and not doing loads of short rows,
> 
> Courseplay has logic to find the direction which results in the lowest number of rows in auto mode, or for the longest, you could use 'longest edge'. But if all your fields are rectangular and oriented to the cardinal directions, it does not really matter...

That's the problem for me I guess and having looked at the example given in #820 I do not play on maps with small fields like in those screenshot examples and don't play a lot of maps with square or rectangle fields, as this is compeltely unnatural to me in a real world example, I also find them very boring to play in game, but hey I'm not here to get into the argument about square fields being unrealistic and rather lazy of map makers. So in my case when playing the game, it's not always easy to see which is the longest edge on a field due to their weird and wonderful shapes and large size. The reason I manually tend to set Courseplay to drive in the cardinal directions of 0/180 or 90/270 is purely because that's what the regular Giants AI worker seems to pick and use due them typically following the limited terrain angles within the map, so I just keep all AI working the same for less collisions and entanglements, when running multiple vehicles on the same fields and performing different tasks at the same time.

**Comment by rohne83** [comment]
I also prefer irregularly shaped large fields, but those rectangular fields are easiest to depict where you have to place the starting point. Coz' the logic where you have to place it to get spirals mode working already today (first half of a 'short edge') remains the same, no matter what size and shape your fields have. Once you understood it for those rectangular fields you're also able to get working positions on irregular fields. Once you've generated the first course, you even know for the weirdest shaped fields which directions the field rows will have, and you can then place the starting point accordingly.
And again as already said by Tensuko and me: spirals is only appropriate for rather narrow fields. On larger fields you should rather go for lands pattern instead (just take care that you set an even number of rows per land in the course generator).

**Comment by daantimmer** [comment]
> I shall have a look at the land pattern next harvest season! I shall update this issue if I get a better/different/other/good result

So, I promised to get back on this, the lands-feature works pretty OK indeed. For me personally it solves my issue. But I see there is some related discussion going on.

**Comment by pvaiko** [comment]
> just take care that you set an even number of rows per land in the course generator

odd numbers should work now (on the main branch and soon in the next release)

**Comment by Thalley** [comment]
Specifying direction for spiral is something that I'd also like. I'm harvesting potatoes with the Ventor 4150 which, as opposed to almost all other harvester, have the pipe on the right side. I've tried to generate a course with spiral that would put the pipe on the non-fruit side, but have not been able to do so. 

The only way I can make it work is by maxing out the number of headlands, but spiral with direction would work much better for that (large) machine.
